internal/server/rewrapping.go

```diff
@@ -0,0 +1,92 @@
+package server
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+)
+
+func init() {
+	kms.RegisterTableRewrapFn("worker_auth_ca_certificate", workerAuthCertRewrapFn)
+	kms.RegisterTableRewrapFn("worker_auth_authorized", workerAuthRewrapFn)
+}
+
+func workerAuthCertRewrapFn(ctx context.Context, dataKeyVersionId string, scopeId string, reader db.Reader, writer db.Writer, kmsRepo *kms.Kms) error {
+	const op = "server.workerAuthCertRewrapFn"
+	var certs []*RootCertificate
+	// indexes on public id, state. neither of which are queryable via scope. this is the fastest query
+	if err := reader.SearchWhere(ctx, &certs, "key_id=?", []interface{}{dataKeyVersionId}, db.WithLimit(-1)); err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	wrapper, err := kmsRepo.GetWrapper(ctx, scopeId, kms.KeyPurposeDatabase)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	newKeyVersionId, err := wrapper.KeyId(ctx)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	for _, cert := range certs {
+		privateKey, err := decrypt(ctx, cert.PrivateKey, wrapper)
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		if cert.PrivateKey, err = encrypt(ctx, privateKey, wrapper); err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		cert.KeyId = newKeyVersionId
```

Comment:
We should probably refactor this to be a `encrypt` and `decrypt` method on the `RootCertificate` type instead. What do you think? I think it would also mean moving these types to use `structwrapping` like the other encrypt and decrypt methods do.

Reply:
I completely agree, but I don't think that is in the context of this ticket or this PR

Reply:
Could we pick it up as a quick follow up to this PR if this PR specifically? I think there's a lot of value in making sure we don't just leave it for the next person to be confused.